### ISSUE TITLE
System.ArgumentException: Cannot generate a hash code for Encounter (Parameter 'x')

### ISSUE THREAD

**Issue by alexzautke** [FirelyTeam/firely-cql-sdk#62]
```
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.ArgumentException: Cannot generate a hash code for Encounter (Parameter 'x')
   at Hl7.Cql.Comparers.CqlComparers.GetHashCode(Object x)
   at Hl7.Cql.Runtime.CqlOperators.GetHashCode(Object x)
   at Hl7.Cql.Comparers.CqlComparerBridge`1.GetHashCode(T obj)
   at System.Collections.Generic.HashSet`1.AddIfNotPresent(T value, Int32& location)
   at System.Collections.Generic.HashSet`1.Add(T item)
   at System.Linq.Enumerable.UnionIterator`1.StoreFirst()
   at System.Linq.Enumerable.UnionIterator`1.MoveNext()
   at System.Linq.Enumerable.CastIterator[TResult](IEnumerable source)+MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Hl7.Cql.Runtime.CqlOperators.ListUnion[T](IEnumerable`1 left, IEnumerable`1 right)
   at AdultOutpatientEncountersFHIR4_2_2_000.Qualifying_Encounters_Value()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at AdultOutpatientEncountersFHIR4_2_2_000.Qualifying_Encounters()
```

**Comment by ewoutkramer** [comment]
Although we seem to have installed converters for every resource type, so the custom hash function should be invoked here, it does not happen. The custom hash function would hash the id, and this should all work.

**Comment by ewoutkramer** [comment]
Fixed by #68.
